app/models/user.rb

```diff
@@ -104,6 +104,11 @@ def access_token_client
     @access_token_client ||= Octokit::Client.new(access_token: access_token, auto_paginate: true) if access_token.present?
   end
 
+  def comment_client(comment)
+    return app_installation_client if app_token.present? && comment.subject.repository.private?
```

Comment:
Shouldn't this be `comment.subject.repository.commentable?`?

Reply:
should the method give you a client to try to post a comment, even if the installation doesn't have write issues. No. Probably not. But it does kinda conflate two concerns. Meh. I'll sort it out.

Reply:
It mostly needs to check if the app is installed on the repository, otherwise any requests with the app_installation_client for that repository will fail